### PR TITLE
Remove special error handling for error not handled in cb which may i…

### PR DESCRIPTION
**Pull request by lincrosenbach**
…gnored nested cb error

### CODE REVIEW

File: src/oop_ext/foundation/callback/_tests/test_callback.py [hunk 1]
```diff
@@ -687,18 +687,18 @@ def BeforeMethod(*args):
...
+        # Now behavior changed and it will fail on first callback error
+        with pytest.raises(RuntimeError):
+            assert c.Method(10) == 20
+
         assert self.before_called == 1
-        assert self.after_called == 1
-        assert mocked.call_count == 2
+        assert self.after_called == 0
+
+        with pytest.raises(RuntimeError):
```
Comment (prusse-martin) [inline src/oop_ext/foundation/callback/_tests/test_callback.py]: This second call and check for raised error is not nescessary.

File: src/oop_ext/foundation/callback/_tests/test_callback.py [hunk 2]
```diff
@@ -1021,3 +1021,40 @@ def Method(a):
...
+                self.c.Register(self._UpdateAValue)
+
+            def SetValue(self, value):
+                self.value = value
+                self.c(value * 0.1)
+
+            def _UpdateAValue(self, new_value):
+                self._a.SetValue(new_value)
+
+        a = B()
```
Comment (nicoddemus) [inline src/oop_ext/foundation/callback/_tests/test_callback.py]: What's this, an obfuscation contest? 😛

Comment (nicoddemus) [inline src/oop_ext/foundation/callback/_tests/test_callback.py]:
```suggestion
        b = B()
```

File: src/oop_ext/foundation/callback/_tests/test_callback.py [hunk 3]
```diff
@@ -1021,3 +1021,40 @@ def Method(a):
...
+        with pytest.raises(ZeroDivisionError):
+            a.SetValue(5)
```
Comment (nicoddemus) [inline src/oop_ext/foundation/callback/_tests/test_callback.py]:
```suggestion
            b.SetValue(5)
```

File: src/oop_ext/foundation/callback/_callback.py [hunk 4]
```diff
@@ -5,51 +5,3 @@
 # ===================================================================================================
 class FunctionNotRegisteredError(RuntimeError):
```
[the remark this answers is not in the document]
Comment (nicoddemus) [reply src/oop_ext/foundation/callback/_callback.py]: I think this is used by `callback.Remove` when the function is not registered.

File: src/oop_ext/foundation/callback/_tests/test_callback.py [hunk 5]
```diff
@@ -1021,3 +1021,40 @@ def Method(a):
         c.Register(partial(Method, "partial"))
         c()
         assert called == ["lambda", "partial"]
+
+    def testCallbackInsideCallback(self):
```
[the remark this answers is not in the document]
Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: Double checked.
Failed: DID NOT RAISE <class 'ZeroDivisionError'>

File: src/oop_ext/foundation/callback/_tests/test_callback.py (hunk 2 above)
Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: :-P

Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: Done

File: src/oop_ext/foundation/callback/_tests/test_callback.py (hunk 3 above)
Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: Done

File: src/oop_ext/foundation/callback/_tests/test_callback.py (hunk 1 above)
Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: Removed

File: src/oop_ext/foundation/callback/_tests/test_callback.py (hunk 2 above)
Comment (lincrosenbach) [reply src/oop_ext/foundation/callback/_tests/test_callback.py]: Renamed

File: src/oop_ext/foundation/callback/_callback.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (prusse-martin) [reply src/oop_ext/foundation/callback/_callback.py]: I dodn't think soo...
https://github.com/ESSS/oop-ext/search?q=FunctionNotRegisteredError&unscoped_q=FunctionNotRegisteredError

[the remark this answers is not in the document]
Comment (nicoddemus) [reply src/oop_ext/foundation/callback/_callback.py]: You are right; we used to raise that exception during `callback.Unregister`, but we changed that to be a no-op and the exception was left there, unused.

Created https://github.com/ESSS/oop-ext/issues/9 to remove it

**Pull request merged**